core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -786,7 +786,7 @@ public boolean isFollowing() {
             DeterministicKey key = (DeterministicKey) entry.getKey();
             Protos.Key.Builder proto = entry.getValue();
             proto.setType(Protos.Key.Type.DETERMINISTIC_KEY);
-            final Protos.DeterministicKey.Builder detKey = proto.getDeterministicKeyBuilder();
+            final Protos.DeterministicKey.Builder detKey = proto.getDeterministicKey().toBuilder();
```

Comment:
```
            if (entries.isEmpty() && isFollowing()) {
                detKey.setIsFollowing(true);
            }
            proto.setDeterministicKey(detKey); //<--- This line is missing below
            if (key.getParent() != null) {
                // HD keys inherit the timestamp of their parent if they have one, so no need to serialize it.
                proto.clearCreationTimestamp();
```

Reply:
This missing line causes about 11 tests to fail and probably has a lot of other side effects when saving the wallet protobuf file.